### ISSUE TITLE
emissive_color - sequences of dimension 0 should contain 4 items, not 5

### ISSUE THREAD

**Issue by fire** [KhronosGroup/glTF-Blender-IO#858]
**Describe the bug**

Blender crashes on file

**To Reproduce**
1. Open glb file
2. See error

**Expected behavior**
Load the glb correctly.

**.blend file/ .gltf**

**Version**
 - OS: linux
 - Blender Version 2.81 and https://github.com/KhronosGroup/glTF-Blender-IO/pull/857

**Additional context**

https://github.com/KhronosGroup/glTF-Blender-IO/blob/2ad329e0fdf89803c09df686ca866ee6782542d0/addons/io_scene_gltf2/blender/imp/gltf2_blender_map_emissive.py#L80-L84

Should be:

```
        else:
            emissive_color = pymaterial.emissive_factor
            if len(emissive_color) == 3:
                emissive_color.append(1.0) # add alpha
            emit.inputs[0].default_value = emissive_color
```
```
Stack Trace:

bpy.context.space_data.recent_folders_active = 0
Traceback (most recent call last):
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/__init__.py", line 919, in execute
    return self.import_gltf2(context)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/__init__.py", line 933, in import_gltf2
    if self.unit_import(path, import_settings) == {'FINISHED'}:
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/__init__.py", line 956, in unit_import
    BlenderGlTF.create(self.gltf_importer)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_gltf.py", line 40, in create
    BlenderScene.create(gltf)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_scene.py", line 43, in create
    BlenderNode.create_vnode(gltf, 'root')
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_node.py", line 45, in create_vnode
    BlenderNode.create_vnode(gltf, child)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_node.py", line 45, in create_vnode
    BlenderNode.create_vnode(gltf, child)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_node.py", line 45, in create_vnode
    BlenderNode.create_vnode(gltf, child)
  [Previous line repeated 1 more time]
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_node.py", line 35, in create_vnode
    BlenderNode.create_object(gltf, vnode_id)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_node.py", line 53, in create_object
    obj = BlenderNode.create_mesh_object(gltf, pynode, name=vnode.name)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_node.py", line 178, in create_mesh_object
    mesh = BlenderMesh.create(gltf, pynode.mesh, pynode.skin)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_mesh.py", line 57, in create
    BlenderMaterial.create(gltf, prim.material, vertex_color)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_material.py", line 96, in create
    BlenderEmissiveMap.create(gltf, material_idx, vertex_color, factor_only=True)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_map_emissive.py", line 30, in create
    BlenderEmissiveMap.create_nodetree(gltf, material_idx, vertex_color, factor_only)
  File "/home/fedora/Documents/blender-tools/addons/io_scene_gltf2/blender/imp/gltf2_blender_map_emissive.py", line 83, in create_nodetree
    emit.inputs[0].default_value = emissive_color
ValueError: bpy_struct: item.attr = val: sequences of dimension 0 should contain 4 items, not 5

location: <unknown location>:-1

```

**Comment by scurest** [comment]
[The spec says](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#materialemissivefactor) emissiveFactor has to be length 3.

**Comment by fire** [comment]
This is the emission texture though...

**Comment by fire** [comment]
I have no idea where the extra element comes from. It's not in the json part.

**Comment by fire** [comment]
Will try to get you a sample.

**Comment by scurest** [comment]
Well `emissive_color.append(1.0)` is mutating the value in the pymaterial in place, so if this is ever called twice it would be length 5 then. It should be doing `emissive_color = pymaterial.emissive_factor + [1]`.

**Comment by fire** [comment]
```python
        else:
            emit.inputs[0].default_value = pymaterial.emissive_factor + [1]
```

Doing a test.
This also works as suggested.

**Comment by scurest** [comment]
Yeah, it really shouldn't change the value like that.

This should be fixed by #831 btw lol.

**Comment by fire** [comment]
Thanks for the help.

![image](https://user-images.githubusercontent.com/32321/71777273-6e258d80-2f52-11ea-8374-5f90c362d8c5.png)

Here's a cool screenshot.

**Comment by scurest** [comment]
Here's a test file: [iss858.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/4023019/iss858.zip)

Two meshes with the same material. One with vertex colors, one without. The pymaterial is visited twice, once to create a node tree with the vertex color node and once without. The second time triggers the error.

Can confirm this is fixed by #831.

**Comment by julienduroure** [comment]
Hello,
Thanks for the investigation.
#831 will not land in 2.82, so I propose that we create a PR to fix only this issue, so this can be in 2.82

**Comment by scurest** [comment]
Ha, I figured. I just thought it was funny :)
